lib/iris/_lazy_data.py

```diff
@@ -138,7 +124,10 @@ def as_concrete_data(data):
         # In some cases dask may return a scalar numpy.int/numpy.float object
         # rather than a numpy.ndarray object.
         # Recorded in https://github.com/dask/dask/issues/2111.
+        dtype = data.dtype
```

Comment:
The implication here is that `array.dtype == array.compute().dtype` is not guaranteed.
Is that really true + if so why ??

Reply:
If the result of the compute is a `MaskedConstant`, it will always have a float dtype